complement/pipeline.yml

```diff
@@ -73,6 +73,53 @@ steps:
         - exit_status: 128
           limit: 3
 
+  - command:
+      # Build the Synapse Workers docker image, which is located at:
+      # https://github.com/matrix-org/synapse/blob/master/docker/Docker-workers
+      # Note that this image is based on matrixdotorg/synapse:latest, which we're just
+      # pulling from docker hub
+      - wget https://github.com/matrix-org/synapse/archive/master.tar.gz
+      - tar -xzf master.tar.gz
+      - docker build -t matrixdotorg/synapse:workers -f synapse-master/docker/Dockerfile-workers synapse-master/
+      # Build the accompanying Synapse workers Complement docker container. This specifically
+      # sets up a Synapse worker setup for Complement, and is located at:
+      # https://github.com/matrix-org/complement/blob/master/dockerfiles/SynapseWorkers.Dockerfile.
+      - docker build -t complement-synapse -f dockerfiles/SynapseWorkers.Dockerfile dockerfiles/
+      # Run the tests!
+      - COMPLEMENT_BASE_IMAGE=complement-synapse go test -v -tags synapse_blacklist ./tests
```

Comment:
why is `COMPLEMENT_BASE_IMAGE` set here, and a bunch of other `COMPLEMENT_*` vars are set in `environment`? consistency ftw.